### DISCUSSION TITLE
Compiling parent when compiling child

### DISCUSSION THREAD

**Discussion by NecroSyri** [glenn2223/vscode-live-sass-compiler#296] [Q&A] [answered]
Is there a way to automatically compile parent when compiling child ?

For exemple :

parent.scss
`@use "child.scss"`

child.scss
`*{background-color:red;}`

If I change child.scss background to blue and save, it will recompile child.scss, I'd like it to automatically recompile parent.scss too, instead of having to do it manually everytime to see the changes, is it possible ?

**Comment by glenn2223** [comment, marked as answer]
This extension doesn't read file contents to identify links. Instead - if a partial file is saved - it triggers the compilation of non-partial files. We also use the SASS principle of "underscored file prefixes are partial files". All this means that `path/_file-starting-with-underscore.scss` would trigger the compilation of all non-partial files.

However, if you have a common directory/format for all your partial files, you can use [`liveSassCompile.settings.partialsList`](https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/docs/settings.md#livesasscompilesettingspartialslist). This setting allows you to specify an array of glob pattern that matches your partial files.

---

Linked to #17

**Reply by NecroSyri** [reply]
Thanks, that worked perfectly

**Comment by ChristopherWelte** [comment]
This is _**incredible**_! I have a rather complicated partials folder structure and so it took some tinkering, but now whenever I save _any_ partial, the main.scss file compiles. Your YouTube video was very helpful. Thank you!

**Reply by glenn2223** [reply]
Wow, thanks @ChristopherWelte - you made my day!

I was unsure about - and, as such, I was very hypercritical of my video - but I hoped that it would help at least one person... and it has! Thanks for all the kind words 😁🥰

**Reply by ChristopherWelte** [reply]
I had been fighting with this other compiler (not going to name names) but it meant that for _every & any_ change to a partial, I would also have to make a minor change to main.scss (like adding a return) so that the file could compile all my partials (I have a LOT). It slowed the process down and was not scalable. Now I'm bringing another coder on our team and this has made onboarding _so much smoother_!
